### PR TITLE
Add a bunch of tracing spans in sparse Merkle writer

### PR DESCRIPTION
**Pull request by AlCutter**
This should give a bit more insight into what's going on inside RPCs to the map server.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly.

### CODE REVIEW

File: merkle/sparse_merkle_tree.go [hunk 1]
```diff
@@ -442,12 +478,16 @@ func (s SparseMerkleTreeReader) BatchInclusionProof(ctx context.Context, rev int
 		}
 		r[string(index)] = ri
 	}
+	span.End()
```
Comment (Martin2112) [inline merkle/sparse_merkle_tree.go]: Might be Github diffs confusing things but does that match the span that's had the End() already deferred?

Comment (AlCutter) [reply merkle/sparse_merkle_tree.go]: should be the one at L456, no?
(I can rename them if you prefer?

Comment (Martin2112) [reply merkle/sparse_merkle_tree.go]: No I see it now. I think renaming just adds noise.

**Comment by Martin2112** [comment]
Looks like some of the tests might be too strict on the context they accept.

**Comment by AlCutter** [comment]
> Looks like some of the tests might be too strict on the context they accept.

Should be fixed now...

**Pull request merged**